nifi-nar-bundles/nifi-azure-bundle/nifi-azure-processors/src/main/java/org/apache/nifi/processors/azure/storage/PutAzureBlobStorage.java

```diff
@@ -0,0 +1,116 @@
+import com.microsoft.azure.storage.blob.CloudBlobContainer;
+
+@Tags({ "azure", "microsoft", "cloud", "storage", "blob" })
+@SeeAlso({ ListAzureBlobStorage.class, FetchAzureBlobStorage.class })
+@CapabilityDescription("Puts content into an Azure Storage Blob")
+@InputRequirement(Requirement.INPUT_REQUIRED)
+@WritesAttributes({ @WritesAttribute(attribute = "azure.container", description = "The name of the Azure container"),
+        @WritesAttribute(attribute = "azure.blobname", description = "The name of the Azure blob"),
+        @WritesAttribute(attribute = "azure.primaryUri", description = "Primary location for blob content"),
+        @WritesAttribute(attribute = "azure.etag", description = "Etag for the Azure blob"),
+        @WritesAttribute(attribute = "azure.length", description = "Length of the blob"),
+        @WritesAttribute(attribute = "azure.timestamp", description = "The timestamp in Azure for the blob")})
+public class PutAzureBlobStorage extends AbstractAzureBlobProcessor {
+
+    public void onTrigger(final ProcessContext context, final ProcessSession session) throws ProcessException {
+        FlowFile flowFile = session.get();
+        if (flowFile == null) {
+            return;
+        }
+
+        final long startNanos = System.nanoTime();
+
+        String containerName = context.getProperty(AzureConstants.CONTAINER).evaluateAttributeExpressions(flowFile).getValue();
+
+        String blobPath = context.getProperty(BLOB).evaluateAttributeExpressions(flowFile).getValue();
+
+        try {
+            CloudStorageAccount storageAccount = createStorageConnection(context, flowFile);
+            CloudBlobClient blobClient = storageAccount.createCloudBlobClient();
+            CloudBlobContainer container = blobClient.getContainerReference(containerName);
+
+            CloudBlob blob = container.getBlockBlobReference(blobPath);
+
+            final Map<String, String> attributes = new HashMap<>();
+            long length = flowFile.getSize();
+            session.read(flowFile, rawIn -> {
+                InputStream in = rawIn;
+                if (!(in instanceof BufferedInputStream)) {
+                    // do not double-wrap
+                    in = new BufferedInputStream(rawIn);
+                }
+
+                try {
+                    blob.upload(in, length);
+                    BlobProperties properties = blob.getProperties();
+                    attributes.put("azure.container", containerName);
+                    attributes.put("azure.primaryUri", blob.getSnapshotQualifiedUri().toString());
+                    attributes.put("azure.etag", properties.getEtag());
+                    attributes.put("azure.length", String.valueOf(length));
+                    attributes.put("azure.timestamp", String.valueOf(properties.getLastModified()));
+                } catch (StorageException | URISyntaxException e) {
+                    throw new IOException(e);
```

Comment:
Same as comment on FetchAzureBlobStorage.  We should either add IOE to below catch block or propagate the actual exception so flow files are properly sent to failure relationship.